pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -8695,6 +8696,99 @@ func TestReconcile_DependencyValidationsImmediatelyFailPipelineRun(t *testing.T)
 	}
 }
 
+// TestReconcileWithResolver checks that a PipelineRun with a populated Resolver
+// field creates a ResourceRequest object for that Resolver's type, and
+// that when the request is successfully resolved the PipelineRun begins running.
+func TestReconcileWithResolver(t *testing.T) {
+	resolverName := "foobar"
+	pr := &v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pr",
+			Namespace: "default",
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			ServiceAccountName: "default",
+			PipelineRef: &v1beta1.PipelineRef{
+				ResolverRef: v1beta1.ResolverRef{
+					Resolver: v1beta1.ResolverName(resolverName),
+				},
+			},
+		},
+	}
+
+	cms := []*corev1.ConfigMap{{
+		ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
+		Data: map[string]string{
+			"enable-api-fields": "alpha",
+		},
+	}}
+
+	d := test.Data{
+		ConfigMaps:   cms,
+		PipelineRuns: []*v1beta1.PipelineRun{pr},
+		ServiceAccounts: []*corev1.ServiceAccount{{
+			ObjectMeta: metav1.ObjectMeta{Name: pr.Spec.ServiceAccountName, Namespace: "foo"},
+		}},
+	}
+
+	prt := newPipelineRunTest(d, t)
+	defer prt.Cancel()
+
+	pipelinerun, _ := prt.reconcileRun("default", "pr", nil, false)
+	cond := pipelinerun.Status.GetCondition(apis.ConditionSucceeded)
+	if cond == nil || cond.Status != corev1.ConditionUnknown || cond.Reason != ReasonResolvingPipelineRef {
+		t.Fatalf("unexpected condition: %#v", cond)
+	}
+
+	client := prt.TestAssets.Clients.ResourceRequests.ResolutionV1alpha1().ResourceRequests("default")
+	resourcerequests, err := client.List(prt.TestAssets.Ctx, metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("unexpected error listing resource requests: %v", err)
+	}
+	numResourceRequests := len(resourcerequests.Items)
+	if numResourceRequests != 1 {
+		t.Fatalf("expected exactly 1 resource request but found %d", numResourceRequests)
+	}
+
+	resreq := &resourcerequests.Items[0]
+	resourceRequestType := resreq.ObjectMeta.Labels["resolution.tekton.dev/type"]
+	if resourceRequestType != resolverName {
+		t.Fatalf("expected resource request type %q but saw %q", resourceRequestType, resolverName)
+	}
+
+	// Mock a successful resolution
+	var pipelineBytes = []byte(`
+          kind: Pipeline
+          apiVersion: tekton.dev/v1beta1
+          metadata:
+            name: foo
+          spec:
+            tasks:
+            - name: task1
+              taskSpec:
+                steps:
+                - name: step1
+                  image: ubuntu
+                  script: |
+                    echo "hello world!"
+        `)
+	resreq.Status.ResourceRequestStatusFields.Data = base64.StdEncoding.Strict().EncodeToString(pipelineBytes)
```

Comment:
Good point, this isn't something I've tackled as part of this PR but I am also starting to think about how the `resolution` package could provide test helpers / mocks for this so that pipelines doesn't have to own them.

Reply:
consider adding a TODO comment here to move out of this test file

Reply:
I created an issue in `resolution` to capture this: https://github.com/tektoncd/resolution/issues/10